tests/components/yessssms/test_notify.py

```diff
@@ -2,6 +2,96 @@
 import unittest
 import requests_mock
 import homeassistant.components.yessssms.notify as yessssms
+from homeassistant.components.yessssms.const import CONF_PROVIDER
+
+from tests.common import get_test_home_assistant
+from homeassistant.const import CONF_PASSWORD, CONF_RECIPIENT, CONF_USERNAME
+
+
+class TestNotifyYesssSMSSetUP(unittest.TestCase):
```

Comment:
New tests must be standalone pytest test functions. See eg:
https://github.com/home-assistant/home-assistant/blob/3efdf29dfa178838002201eaea258db78e6fa2ba/tests/components/manual/test_alarm_control_panel.py#L83-L119

Reply:
just when I finally thought I would be done...

Would you point me to the docs for how to write tests please?

Reply:
Read the pytest docs and eg this blog:
https://realpython.com/python-testing/

https://docs.pytest.org/en/latest/

Mocking is also essential to know sooner or later:
https://docs.python.org/3/library/unittest.mock.html

Then copy paste from existing tests, like the example I referenced.

Reply:
what's the reasoning behind separate functions vs another test-class?

is it the fact that those functions can be async?

Reply:
That's one reason. Another is that there's no reason to use classes. Setup and teardown code should be done in pytest fixtures, which are also separate functions.